Classes/BITChannel.m

```diff
@@ -123,59 +128,89 @@ - (BOOL)isQueueBusy {
   return self.channelBlocked;
 }
 
-- (void)persistDataItemQueue {
+- (void)persistDataItemQueue:(char **)eventBuffer {
   [self invalidateTimer];
-  if (!BITSafeJsonEventsString || strlen(BITSafeJsonEventsString) == 0) {
+  
+  // Make sure string (which points to BITTelemetryEventBuffer) is not changed.
+  char *previousBuffer = NULL;
+  char *newEmptyString = NULL;
+  do {
+    newEmptyString = strdup("");
```

Comment:
Memory leak :worried:

Reply:
Merged the PR that fix... but I'm pretty sure we had fixed this in a prev. PR months ago. =(